### PR TITLE
Plot_ppc - KeyError: "['y'] var names are not present in dataset"

### LINKED ISSUE #843: plot_ppc - KeyError: "['y'] var names are not present in dataset"
Hi, I've had troubles using `plot_ppc` in a project I'm working on, so I tried to reproduce it using the quickstart pystan example. This is the code I'm using:

```python
schools_code = """
data {
  int<lower=0> J;
  real y[J];
  real<lower=0> sigma[J];
}

parameters {
  real mu;
  real<lower=0> tau;
  real theta[J];
}

model {
  mu ~ normal(0, 5);
  tau ~ cauchy(0, 5);
  theta ~ normal(mu, tau);
  y ~ normal(theta, sigma);
}
generated quantities {
    vector[J] log_lik;
    vector[J] y_hat;
    for (j in 1:J) {
        log_lik[j] = normal_lpdf(y[j] | theta[j], sigma[j]);
        y_hat[j] = normal_rng(theta[j], sigma[j]);
    }
}
"""

schools_dat = {'J': 8,
               'y': [28,  8, -3,  7, -1,  1, 18, 12],
               'sigma': [15, 10, 16, 11,  9, 11, 10, 18]}

sm = pystan.StanModel(model_code=schools_code, verbose=False)
fit = sm.sampling(data=schools_dat, iter=1000, chains=4)

data = az.from_pystan(posterior=fit,
                      posterior_predictive='y_hat',
                      observed_data=['y'],
                      log_likelihood='log_lik',
                      coords={'school': schools},
                      dims={'theta': ['school'], 'y': ['school'], 'log_lik': ['school'], 'y_hat': ['school'], 'theta_tilde': ['school']})

az.plot_ppc(data)
```

This raises:

```python
KeyError: "['y'] var names are not present in dataset"
```

which is exactly the kind of error I got in my own project... I guess I'm doing something wrong, here, but I can't figure out what it is.

Any help would be appreciated... thanks!

### PR DESCRIPTION
**Pull request by amukh18**
This fixes #843

### CODE REVIEW

File: arviz/plots/ppcplot.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def plot_ppc(
 
         >>> import arviz as az
         >>> data = az.load_arviz_data('radon')
-        >>> az.plot_ppc(data)
+        >>> az.plot_ppc(data,data_pairs={"obs":"obs"})
```
Comment (ahartikainen) [inline arviz/plots/ppcplot.py]: This looks good. Maybe we could add an example from `pystan` case too, e.g. comment it out
```
>>> # az.plot_ppc(data, data_pairs={"obs" : "obs_hat"})
```
cc @OriolAbril

Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: Also, maybe we could add 4 spaces to indent the lines 65, 66 and make key and value lowercase.

```
Dictionary structure:
    key = data var_name
    value = posterior predictive var_name
```

Comment (amukh18) [reply arviz/plots/ppcplot.py]: I have updated the file with the required lines and changes. However, I am a little unfamiliar with the parameters `"obs"` and `"obs_hat"`. Could you please link me to documentation that may help me learn more about these?

**Comment by OriolAbril** [comment]
LGTM, we can merge after pylint passes. Thanks for your PR!

File: arviz/plots/ppcplot.py [hunk 2]
```diff
@@ -127,7 +127,9 @@ def plot_ppc(
...
+        >>> #az.plot_ppc(data,data_pairs={"obs":"obs_hat"})
+
```
Comment (ahartikainen) [inline arviz/plots/ppcplot.py]: I think there is some white space here, so I think you need to remove it.

File: arviz/plots/ppcplot.py [hunk 3]
```diff
@@ -127,7 +127,9 @@ def plot_ppc(
 
         >>> import arviz as az
         >>> data = az.load_arviz_data('radon')
-        >>> az.plot_ppc(data)
+        >>> az.plot_ppc(data,data_pairs={"obs":"obs"})
+        >>> #az.plot_ppc(data,data_pairs={"obs":"obs_hat"})
```
Comment (canyon289) [inline arviz/plots/ppcplot.py]: This commented out line should be deleted as well

Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: That was my suggestion, do you have other way to include that example?

File: arviz/plots/ppcplot.py (hunk 2 above)
Comment (amukh18) [reply arviz/plots/ppcplot.py]: @ahartikainen Do you mean line 132 in my commit? But I feel that space was there in the form of line 131 in the original file

**Comment by amukh18** [comment]
@OriolAbril I saw the error in the Travis CI build logs and then ran pylint. However, the code passed the checks on my local machine.

File: arviz/plots/ppcplot.py (hunk 2 above)
Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: You just probably need to remove trailing whitespace

File: arviz/plots/ppcplot.py (hunk 3 above)
Comment (OriolAbril) [reply arviz/plots/ppcplot.py]: I would leave the comment here as a remainder of including PyStan examples

File: arviz/plots/ppcplot.py (hunk 2 above)
Comment (OriolAbril) [reply arviz/plots/ppcplot.py]: @amukh18 It looks like you have some spaces (probably 12) in line 132, you should remove them and leave only the new line

**Comment by amukh18** [comment]
Please ignore this last commit.

**Comment by ahartikainen** [comment]
You can edit your commit history with

    git log --oneline

    copy suitable hash

    git rebase -i hashhere

And then drop wrong commit and save and close editor (e.g. vim)

**Comment by amukh18** [comment]
I typed `drop b07e3eb Add files via upload` and closed my editor. However, on typing `git rebase --oneline` , the history about that commit was still there.

**Comment by ahartikainen** [comment]
`git rebase continue`?

**Comment by amukh18** [comment]
I get `No rebase in progress?`

File: arviz/plots/ppcplot.py [hunk 4]
```diff
@@ -127,9 +127,10 @@ def plot_ppc(
...
 
-    Plot the overlay with empirical CDFs.
+     Plot the overlay with empirical CDFs.
```
Comment (OriolAbril) [inline arviz/plots/ppcplot.py]: I think this extra space is what makes the doc build fail

**Pull request merged**